experiments/deployment/waku/experiments/regression.py

```diff
@@ -0,0 +1,176 @@
+        logger.info("Applying bootstrap")
+        kubectl_apply(bootstrap, namespace=namespace)
+        logger.info("bootstrap applied. Waiting for rollout.")
+        wait_for_rollout(bootstrap["kind"], bootstrap["metadata"]["name"], namespace, 2000)
+
+        num_nodes = nodes["spec"]["replicas"]
+        messages = self.get_flag_value("messages", publisher["spec"]["containers"][0]["command"])
+        delay_seconds = self.get_flag_value(
+            "delay-seconds", publisher["spec"]["containers"][0]["command"]
+        )
+
+        # Apply nodes configuration
+        logger.info("Applying nodes")
+        kubectl_apply(nodes, namespace=namespace)
+        logger.info("nodes applied. Waiting for rollout.")
+        timeout = num_nodes * 3000
+        wait_for_rollout(nodes["kind"], nodes["metadata"]["name"], namespace, timeout)
+
+        # TODO [metadata output]: log publish message start time
+        # Apply publisher configuration
+        logger.info("applying publisher")
+        kubectl_apply(publisher, namespace=namespace)
+        logger.info("publisher applied. Waiting for rollout.")
+        wait_for_rollout(
+            publisher["kind"],
+            publisher["metadata"]["name"],
+            namespace,
+            20,
+            api_client,
+            ("Ready", "True"),
+            # TODO [extend condition checks] lambda cond : cond.type == "Ready" and cond.status == "True"
+        )
+        logger.info("publisher rollout done.")
+
+        timeout = num_nodes * messages * delay_seconds * 120
+        logger.info(f"Waiting for Ready=False. Timeout: {timeout}")
+
+        wait_for_rollout(
+            publisher["kind"],
+            publisher["metadata"]["name"],
+            namespace,
+            timeout,
+            api_client,
+            ("Ready", "False"),
+        )
+        # TODO: consider state.reason == .completed
+        time.sleep(20)
+        # TODO [metadata output]: log publish message end time
+
+    def get_cleanup(
+        self, api_client: ApiClient, namespace: str, deployments: List[yaml.YAMLObject]
+    ) -> Callable[[], None]:
+        def cleanup():
+            logger.info("Cleaning up resources.")
+            resources_to_cleanup = get_cleanup_resources(deployments)
+            logger.info(f"Resources to clean up: `{resources_to_cleanup}`")
+
+            logger.info("Start cleanup.")
+            cleanup_resources(resources_to_cleanup, namespace, api_client)
+            logger.info("Waiting for cleanup.")
+            wait_for_cleanup(resources_to_cleanup, namespace, api_client)
+            logger.info("Finished cleanup.")
+
+        return cleanup
+
+    def get_flag_value(self, flag: str, command: List[str]) -> int:
+        for node in command:
+            matches = re.search(f"--{flag}=(?P<numMessages>\\d+)", node)
+            try:
+                return int(matches["numMessages"])
+            except (TypeError, IndexError):
+                pass
+        return None
```

Comment:
Same as before, this could be used in different places. Also, missing `Optional` in return typing.